src/process_utility.c

```diff
@@ -37,6 +37,59 @@ static ProcessUtility_hook_type prev_ProcessUtility_hook;
 
```

Comment:
The reason I had explicit conversions from Cstrings to Name is to be safe in cases the cstring were longer than NAMEDATALEN. In which case, we could get a segfault or memory corruption. Are you sure eliding that is safe in all cases here?

Reply:
I don't see this happening since it is a '\0' terminated string and if it is longer than NAMEDATALEN it would be truncated whenever copied into a NameData. When storing NAMEs into tuples it might make sense to do this but I don't see this being the case here. We are simply calling functions passing a pointer to a ('\0-terminated) string (in fact, both `NameGetDatum()` and `CStringGetDatum()`maps to `PointerGetDatum()`.

We can test this I guess.

Reply:
Also, I am wondering why this is important here as opposed to a lot of other places in the code that passes around strings? What makes this case special?

Reply:
Sorry the case I am referring to is converting char * to a NAME. Do we have a guarantee that the char * is not bigger than NAMEDATALEN. An example is the column_name passed in to process_change_hypertable_column_type.

Reply:
I don't think this is an issue for this case, unless someone is doing an unsafe `strcpy()` somewhere. All NAME-related functions do proper '\0' terminated string manipulation and truncation to NAMELEN, so even if a string longer than NAMEDATALEN is masking itself as a NAME it should be OK. 

However, I anyhow pushed an update that should address your concerns. All C strings are passed through `namein()` in the macro before the upcall.